packages/webcomponents/src/components/business-form-stepped/business-core-info/business-core-info-form-step.tsx

```diff
@@ -23,17 +28,81 @@ import { CoreBusinessInfo, ICoreBusinessInfo } from '../../../api/Business';
   styleUrl: 'business-core-info-form-step.scss',
 })
 export class BusinessCoreInfoFormStep {
-  @Prop() formController: FormController;
+  @Prop() authToken: string;
+  @Prop() businessId: string;
+  @State() formController: FormController;
   @State() errors: any = {};
-  @State() genericInfo: ICoreBusinessInfo = {};
+  @State() coreInfo: ICoreBusinessInfo = {};
+  @Event({ bubbles: true }) submitted: EventEmitter<{ data?: any}>;
+  @Event({ bubbles: true }) formLoading: EventEmitter<boolean>;
+  @Event() serverError: EventEmitter<{ data?: any, message?: string }>;
 
   constructor() {
     this.inputHandler = this.inputHandler.bind(this);
+    this.sendData = this.sendData.bind(this);
+    this.fetchData = this.fetchData.bind(this);
+  }
+
+  private api: any;
+
+  get businessEndpoint() {
+    return `entities/business/${this.businessId}`
+  }
+
+  private async fetchData() {
+    this.formLoading.emit(true);
+    try {
+      const response: IApiResponse<IBusiness> = await this.api.get(this.businessEndpoint);
+      this.coreInfo = new CoreBusinessInfo(response.data);
+      this.formController.setInitialValues({ ...this.coreInfo });
+    } catch (error) {
+      this.serverError.emit({ data: error, message: 'Error fetching business data' });
+    } finally {
+      this.formLoading.emit(false);
+    }
+  }
+
+  private async sendData(onSuccess?: () => void) {
+    this.formLoading.emit(true);
+    try {
+      const payload = parseCoreInfo(flattenNestedObject(this.formController.values.getValue()));
+      const response = await this.api.patch(this.businessEndpoint, JSON.stringify(payload));
+      this.handleResponse(response, onSuccess);
+    } catch (error) {
+      this.serverError.emit({ data: error, message: 'Error updating business data' });
+    } finally {
+      this.formLoading.emit(false);
+    }
+  }
+  
+  handleResponse(response, onSuccess) {
+    if (response.error) {
+      this.serverError.emit({ data: response.error, message: 'Error updating business data' });
+    } else {
+      onSuccess();
+    }
+    this.submitted.emit({ data: response });
```

Comment:
Instead of having `this.serverError` and `this.submitted`, can we have JUST `this.submitted` and always `emit` the full response instead of formatting it differently? This way we don't have to worry about any inconsistencies, it's just always the response from the server

Reply:
We can absolutely change it - 

However here is my logic for doing it this way: 

`this.serverError` is NOT emitted all the way to the customer. My intention with emitting it as an event this way is only for the parent component (`business-form-stepped`) to listen to it from the child components and trigger an Error state in the parent component. 

`this.submitted` is still the event that is emitted to the customer / dev from the sub-components, and it gets emitted regardless of a successful response OR an error response. 

Perhaps it is confusing and unnecessary to emit the full error object / data in the event, and we could simply emit the string that says `Failure to get Business Data` or `Failure to update business data`. 

What do you think?

Reply:
Discussed this again in standup this morning, we are going to use the event emitter logic here to trigger the form alert in the parent form component.